### PR TITLE
SAA-107 introducing new endpoint for rollout prisons.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/RolloutPrisonRepository.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.repository
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.entity.RolloutPrison
+
+@Service
```
Comment (markford-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/RolloutPrisonRepository.kt:7]: Just a question - I would notmally expect to see the @Repository annotation in the data layer as opposed to the @Service annotation. I know they are pretty much synonyms but I think the semantics are useful and also i believe there are some differences in exception translation

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/RolloutPrisonController.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.model.Activity
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.model.RolloutPrison
+import uk.gov.justice.digital.hmpps.hmppsactivitiesmanagementapi.service.RolloutPrisonService
+
+// TODO add pre-auth annotations to enforce roles when we have them
+
+@RestController
+@RequestMapping("/rolloutPrisons", produces = [MediaType.APPLICATION_JSON_VALUE])
```
Comment (markford-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/RolloutPrisonController.kt:22]: Is it a convention that we will use camel case for uris as opposed to rollout-prisons ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/TransformFunctions.kt [hunk 3]
```diff
@@ -138,3 +140,10 @@ private fun EntityActivityPay.toModelActivityPay() =
     pieceRate = this.pieceRate,
     pieceRateItems = this.pieceRateItems
   )
+
+fun transform(prison: EntityRolloutPrison) = ModelRolloutPrison(
```
Comment (markford-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/TransformFunctions.kt:144]: Is it worth looking at a mapping library such as dozer or mapstruct ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/RolloutPrisonRepository.kt (hunk 1 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/repository/RolloutPrisonRepository.kt:7]: Doh.  This is a mistake on my part.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/RolloutPrisonController.kt (hunk 2 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/RolloutPrisonController.kt:22]: No you are right it should have been hyphenated.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/TransformFunctions.kt (hunk 3 above)
Comment (matt-crowson-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/TransformFunctions.kt:144]: So this is purely based on the pattern copied over from CVL project. Perhaps talk about it.

**Review by markford-moj — APPROVED** [review]

**Pull request merged**